tests/Resources/buffer.addontest.js

```diff
@@ -0,0 +1,1647 @@
+			should(buf.writeUInt16LE(0xdead, 0)).eql(2);
+			should(buf.writeUInt16LE(0xbeef, 2)).eql(4);
+			should(buf).eql(Buffer.from([ 0xad, 0xde, 0xef, 0xbe ]));
+		});
+
+		it('throws when trying to write offset out of range', () => {
+			const buf = Buffer.from([ 1, -2 ]);
+			should.throws(() => {
+				buf.writeUInt16LE(2, 2);
+			}, RangeError);
+		});
+
+		it('throws when trying to write values out of range', () => {
+			const buf = Buffer.from([ 1, 2 ]);
+			should.throws(() => {
+				buf.writeUInt16LE(-25243, 0);
+			}, RangeError);
+			should.throws(() => {
+				buf.writeUInt16LE(65536, 0);
+			}, RangeError);
+		});
+	});
+
+	describe('#writeUInt32BE()', () => {
+		it('is a function', () => {
+			const buf = Buffer.from('test');
+			should(buf.writeUInt32BE).be.a.Function;
+		});
+
+		it('handles good value and offset', () => {
+			const buf = Buffer.allocUnsafe(4);
+			should(buf.writeUInt32BE(0xfeedface, 0)).eql(4);
+			should(buf).eql(Buffer.from([ 0xfe, 0xed, 0xfa, 0xce ]));
+		});
+
+		it('throws when trying to write offset out of range', () => {
+			const buf = Buffer.from([ 1, -2 ]);
+			should.throws(() => {
+				buf.writeUInt32BE(2, 2);
+			}, RangeError);
+		});
+
+		it('throws when trying to write values out of range', () => {
+			const buf = Buffer.from([ 1, 2 ]);
+			should.throws(() => {
+				buf.writeUInt32BE(-3, 0);
+			}, RangeError);
+			should.throws(() => {
+				buf.writeUInt32BE(4294967297, 0);
+			}, RangeError);
+		});
+	});
+
+	describe('#writeUInt32LE()', () => {
+		it('is a function', () => {
+			const buf = Buffer.from('test');
+			should(buf.writeUInt32LE).be.a.Function;
+		});
+
+		it('handles good value and offset', () => {
+			const buf = Buffer.allocUnsafe(4);
+			should(buf.writeUInt32LE(0xfeedface, 0)).eql(4);
+			should(buf).eql(Buffer.from([ 0xce, 0xfa, 0xed, 0xfe ]));
+		});
+
+		it('throws when trying to write offset out of range', () => {
+			const buf = Buffer.from([ 1, -2 ]);
+			should.throws(() => {
+				buf.writeUInt32LE(2, 2);
+			}, RangeError);
+		});
+
+		it('throws when trying to write values out of range', () => {
+			const buf = Buffer.from([ 1, 2 ]);
+			should.throws(() => {
+				buf.writeUInt32LE(-2, 0);
+			}, RangeError);
+			should.throws(() => {
+				buf.writeUInt32LE(4294967296, 0);
+			}, RangeError);
+		});
+	});
+
+	describe('#writeUIntBE()', () => {
+		it('is a function', () => {
+			const buf = Buffer.from('test');
+			should(buf.writeUIntBE).be.a.Function;
+		});
+
+		it('handles good value and offset', () => {
+			const buf = Buffer.allocUnsafe(6);
+			should(buf.writeUIntBE(0x1234567890ab, 0, 6)).eql(6);
+			should(buf).eql(Buffer.from([ 0x12, 0x34, 0x56, 0x78, 0x90, 0xab ]));
+		});
+
+		it('throws when trying to write offset out of range', () => {
+			const buf = Buffer.from([ 1, -2 ]);
+			should.throws(() => {
+				buf.writeUIntBE(0x1234567890ab, 1, 6);
+			}, RangeError);
+		});
+
+		// it('throws when trying to write values out of range', () => {
```

Comment:
Same here, can you add a fixme note and explain why this is commented out?

Reply:
I'll add it back and get it right. I got a bit fatigued of writing all these tests!